components/seldon-request-logger/app/default_logger.py

```diff
@@ -135,11 +135,19 @@ def process_and_update_elastic_doc(
 
             no_items_in_batch = len(new_content_part["instance"])
             index = 0
-            for item in new_content_part["instance"]:
+            elements = None
+            if "elements" in new_content_part:
+                elements = new_content_part["elements"]
+
+            for num, item in enumerate(new_content_part["instance"],start=0):
 
                 item_body = doc_body.copy()
 
                 item_body[message_type]["instance"] = item
+
+                if type(elements) == type([]) and len(elements) >= num:
```

Comment:
what is this condition for length `>= num`?

Reply:
Oh, I think it should be >. It is checking that elements array contains an entry at this index. So if at index 2 then elements needs to be of length of at least three. I will try changing it.